skimage/measure/tests/test_marching_cubes.py

```diff
@@ -74,5 +97,58 @@ def test_correct_mesh_orientation():
     np.testing.assert_array_equal(expected, corrected_faces1)
 
 
+def test_both_algs_same_result_ellipse():
+    # Performing this test on data that does not have ambiguities
+    
+    sphere_small = ellipsoid(1, 1, 1, levelset=True)
+    
+    vertices1, faces1, *_ = marching_cubes(sphere_small, 0)
+    vertices2, faces2, *_ = marching_cubes_lewiner(sphere_small, 0, allow_degenerate=False)
+    vertices3, faces3, *_ = marching_cubes_lewiner(sphere_small, 0, allow_degenerate=False, use_classic=True)
+    
+    # Order is different, best we can do is test equal shape and same vertices present
+    assert _same_mesh(vertices1, faces1, vertices2, faces2)
+    assert _same_mesh(vertices1, faces1, vertices3, faces3)
+
+
+def _same_mesh(vertices1, faces1, vertices2, faces2):
+    rounder = lambda x: int(x*1000)/1000  # to take into account small variations
+    triangles1 = vertices1[np.array(faces1)]
+    triangles2 = vertices2[np.array(faces2)]
+    triang1 = [np.concatenate(sorted(t, key=lambda x:tuple(x))) for t in triangles1]
+    triang1 = set([tuple([rounder(i) for i in t]) for t in triang1])
+    triang2 = [np.concatenate(sorted(t, key=lambda x:tuple(x))) for t in triangles2]
+    triang2 = set([tuple([rounder(i) for i in t]) for t in triang2])
+    return triang1 == triang2
+
+def test_both_algs_same_result_donut():
+    # Performing this test on data that does not have ambiguities
+    
+    n = 48
+    a, b = 2.5/n, -1.25
+    isovalue = 0.0
+    #
+    vol = np.empty((n,n,n), 'float32')
+    for iz in range(vol.shape[0]):
+        for iy in range(vol.shape[1]):
+            for ix in range(vol.shape[2]):
+                z, y, x = float(iz)*a+b, float(iy)*a+b, float(ix)*a+b
+                vol[iz,iy,ix] = ( ( 
+                    (8*x)**2 + (8*y-2)**2 + (8*z)**2 + 16 - 1.85*1.85 ) * ( (8*x)**2 +
```

Comment:
Sorry, it has been a while. I am pretty sure it's from Lewiner's code somewhere, but I cannot recall nor find the exact source.